_source/_posts/2022-04-06-react-azure-functions.md

```diff
@@ -0,0 +1,416 @@
+
+This tutorial will show you how to create a React application and deploy it to Azure Static Web Apps. This application will be on the Azure free tier, so you will not be able to rely on the built-in authentication providers that connect Azure and Okta to handle the authentication. Therefore you will use the `okta-react` package from Okta to secure your SPA manually. Once the user authenticates, they may upload an image and receive a badge from a serverless Azure function.
+
+This function will handle the work of accepting the inputted image from the SPA and using a template to create a personalized badge for the user. Since you will be using the free version of Azure Static Web Apps, you will have to deploy the Function as a _Managed Azure Function_.
+
+You will write your application in Visual Studio Code and use the Azure extensions for Functions and Static Web Apps.
+
+**Prerequisites**
+
+- [Node.js](https://nodejs.org/en/)
+- [Azure Account](https://azure.microsoft.com/en-us/)
+- [GitHub Account](https://www.github.com/)
+- [Okta CLI](https://cli.okta.com)
+  > [Okta](https://developer.okta.com/) has Authentication and User Management APIs that reduce development time with instant-on, scalable user infrastructure. Okta's intuitive API and expert support make it easy for developers to authenticate, manage and secure users and roles in any application.
+- [Visual Studio Code](https://code.visualstudio.com/)
+  - [Azure Functions VS Code Extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurefunctions) 
+  - [Azure Static Web Apps VS Code Extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurestaticwebapps)
+
+If you want to see the code, you can download it or fork it [on github](https://github.com/oktadev/okta-react-azure-functions-example).
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+## Create your Okta application
+
+{% include setup/cli.md type="spa" framework="React" loginRedirectUri="http://localhost:4280/login/callback" %}
+
+## Create your React application
+
+The next step is to build your React application as your static web app. Begin as you would with most react apps by running `npx create-react-app azure-static-app`. After a few moments, your application will be ready to run. Once this happens, delete the `.gitignore` file and the `.git` folder that `create-react-app` produced. At this time, there is no way to prevent the task from adding these, but they will conflict with the Azure git files you will add soon.
```

Comment:
```suggestion
The next step is to build your React application as a static web app. Begin as you would with most React apps by running `npx create-react-app azure-static-app`. After a few moments, your application will be ready to run. Once this happens, delete the `.gitignore` file and the `.git` folder that `create-react-app` produced. At this time, there is no way to prevent the task from adding these, but they will conflict with the Azure git files you will add soon.
```
clarity